### PR TITLE
[BugFix] Allow qualify and where to exist at the same time

### PR DESCRIPTION
**Pull request by choury**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Allow qualify and where to exist at the same time, like this
```
select tc from tall qualify dense_rank() OVER(PARTITION by ta order by tg) > 1 where tb <> 'tc'
```
## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by kangkaisen — DISMISSED** [review]

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/ExpressionTest.java [hunk 1]
```diff
@@ -1249,6 +1249,11 @@ public void testQualifyForWindowFunction() throws Exception {
         plan = getFragmentPlan(sql);
         assertContains(plan, "predicates: 11: dense_rank() > 1");
 
+        // for where
+        sql = "select tc from tall qualify dense_rank() OVER(PARTITION by ta order by tg) > 1 where tb <> 'tb'";
+        plan = getFragmentPlan(sql);
+        assertContains(plan, "predicates: 11: dense_rank() > 1");
```
Comment (packy92) [inline fe/fe-core/src/test/java/com/starrocks/sql/plan/ExpressionTest.java:1255]: Could you add the `tb <> 'tb'` info in the planFragment to this test? Then we can see the effect of this pr.

**Review by packy92 — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4 [hunk 2]
```diff
@@ -1587,7 +1587,7 @@ querySpecification
     : SELECT setVarHint* setQuantifier? selectItem (',' selectItem)*
       fromClause
       ((QUALIFY qualifyFunction=selectItem comparisonOperator limit=INTEGER_VALUE)?
-      | (WHERE where=expression)? (GROUP BY groupingElement)? (HAVING having=expression)?)
+       (WHERE where=expression)? (GROUP BY groupingElement)? (HAVING having=expression)?)
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:1590]: qualify should be placed after having

**Review by kangkaisen — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**